apps/balancer-tools/src/app/apr/round/(components)/PoolListTable.tsx

```diff
@@ -112,16 +137,26 @@ export function PoolListTable({ roundId }: { roundId: string }) {
           </Table.HeaderCell>
         </Table.HeaderRow>
         <Table.Body>
-          {tableData.map((gauge) => (
-            <TableRow
-              tableData={tableData}
-              setTableData={setTableData}
-              key={gauge.id}
-              poolId={gauge.id}
-              network={gauge.network}
-              roundId={roundId}
-            />
-          ))}
+          {isLoadingFetchPoolList ? (
+            <Table.BodyRow>
+              <Table.BodyCell colSpan={4}>
+                <div className="flex justify-center">
+                  <Spinner />
+                </div>
+              </Table.BodyCell>
+            </Table.BodyRow>
+          ) : (
+            tableData.map((gauge) => (
+              <TableRow
+                tableData={tableData}
+                setTableData={setTableData}
+                key={gauge.id}
+                poolId={gauge.id}
+                network={gauge.network}
+                roundId={roundId}
+              />
+            ))
```

Comment:
You can move the TableRow component to another file and put a Suspense boundary around it